ml/backend/ggml/ggml.go

```diff
@@ -596,10 +596,13 @@ func (t *Tensor) View(ctx ml.Context, offset int, shape ...int) ml.Tensor {
 }
 
 const (
-	ropeTypeNorm C.int = iota
+	ropeTypeNorm   C.int = 0
+	ropeTypeNeox   C.int = 2
+	ropeTypeMrope  C.int = 8
+	ropeTypeVision C.int = 24
 )
 
-func (t *Tensor) RoPE(ctx ml.Context, positionIDs, ropeFactors ml.Tensor, ropeDim uint32, ropeBase, ropeScale float32) ml.Tensor {
+func (t *Tensor) RoPE(ctx ml.Context, positionIDs, ropeFactors ml.Tensor, ropeDim, ropeType uint32, ropeBase, ropeScale float32) ml.Tensor {
```

Comment:
I took a similar but slightly different approach here when dealing with rope settings doing qwen2. Broke it off into a separate PR here:
https://github.com/ollama/ollama/pull/9121

Please take a look! The main difference is that I added a new struct for RoPE config since the function parameters were getting out of hand here.Along with allowing specifying n_context_train.